src/adapter/etl-adapter-google-sheet/src/Flow/ETL/Adapter/GoogleSheet/GoogleSheetExtractor.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Flow\ETL\Adapter\GoogleSheet;
+
+use Flow\ETL\DSL\Entry;
+use Flow\ETL\Extractor;
+use Flow\ETL\FlowContext;
+use Flow\ETL\Row;
+use Flow\ETL\Rows;
+use Google\Client;
+use Google\Exception;
+use Google\Service\Sheets;
+use Webmozart\Assert\Assert;
+
+final class GoogleSheetExtractor implements Extractor
+{
+    public function __construct(
+        private readonly Sheets $service,
+        private readonly string $spreadsheetId,
+        private readonly GoogleSheetRange $initialDataRange,
+        private readonly bool $withHeader,
+        private readonly int $rowsInBatch,
+    ) {
+        Assert::greaterThan($rowsInBatch, 0);
+    }
+
+    /**
+     * @throws Exception
+     * @throws \JsonException
+     */
+    public static function create(string $authJson, string $spreadsheetId, GoogleSheetRange $initialDataRange, bool $withHeader = true, int $rowsInBatch = 500) : self
```

Comment:
`string $authJson` why do you pass `json-string` as a parameter instead of an `array` or a configuration object? Doesn't passing `json-string` make DX a nightmare?